### ISSUE TITLE
surface.DrawText() breaking if getting nil

### ISSUE THREAD

**Issue by eLeCtrOssSnake** [Facepunch/garrysmod-issues#3750]
### Details
A simple code snippet to reproduce game breaking(client)
cam.Start3D2D(LocalPlayer():GetPos(), LocalPlayer():GetAngles(), 1)
	surface.DrawText(nil)
cam.End3D2D()

there is no difference if you set parameters before drawing nil.
I found it working everywhere. And if you start acting with broken text, like make prop(or ent) transparent it will break even more with a flipping world on 90 degrees.

### Steps to reproduce
Make support of drawing nil or at least make a check for nil, and don't try to draw it as it is.

**Comment by Kefta** [comment]
That's because nil isn't a valid argument to DrawText and you are causing an error before the cam can be ended. You could add a bunch of safeguards by running your code in an environment that maintains your own render stack counts and pops them whenever errors occur, but I don't think a fix will be made to GMod itself.

**Comment by eLeCtrOssSnake** [comment]
Nope. It ends cam.Start3D2D() properly. It breaks in a different way. running cam.End3D2D() does not do anything. It breaks only when trying to draw nil as a text value. I think it should be fixed some way.

**Comment by Kefta** [comment]
Do you have any code where sending in nil to DrawText works? Any video of the issue? Everywhere I try I get this error:
```
[ERROR] lua/autorun/client/jumpman.lua:3: bad argument #1 to 'DrawText' (string expected, got nil)
  1. DrawText - [C]:-1
   2. v - lua/autorun/client/jumpman.lua:3
    3. unknown - lua/includes/modules/hook.lua:84
```

**Comment by eLeCtrOssSnake** [comment]
After a couple of tests, I found that game breaks in different ways. Some world flip is caused by not ending cam2d3d. And can be fixed for one time running ending of it. But there is another kind of render breaking when world getting flipped by 300 degrees or something about it and it starts rendering kinda like chunks in minecraft. This happens more or less randomly. When I first discovered it after trying to make entity in dark rp it was breaking like minecraft style only(was in draw hook on client so it was called a lot). In one time test it probably breaks map renderer still not closing cam2d3d but if it getting called one more frame it will break everything completely. When you end2d3d and normals are back, world is still broken. I still think it should be fixed.
https://streamable.com/6j0la
https://streamable.com/bki43
